Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -387,14 +387,23 @@ static Ref<CSSPrimitiveValue> textSpacingTrimFromStyle(const RenderStyle& style)
     return CSSPrimitiveValue::create(CSSValueSpaceAll);
 }
 
-static Ref<CSSPrimitiveValue> textAutospaceFromStyle(const RenderStyle& style)
+static Ref<CSSValue> textAutospaceFromStyle(const RenderStyle& style)
 {
     // FIXME: add support for remaining values once spec is stable and we are parsing them.
     auto textAutospace = style.textAutospace();
     if (textAutospace.isAuto())
         return CSSPrimitiveValue::create(CSSValueAuto);
+    if (textAutospace.isNoAutospace())
+        return CSSPrimitiveValue::create(CSSValueNoAutospace);
+    if (textAutospace.isNormal())
+        return CSSPrimitiveValue::create(CSSValueNormal);
+
+    if (textAutospace.hasIdeographAlpha())
+        return CSSPrimitiveValue::create(CSSValueIdeographAlpha);
+    if (textAutospace.hasIdeographNumeric())
+        return CSSPrimitiveValue::create(CSSValueIdeographNumeric);
 
```

Comment:
It's interesting because `ideograph-alpha ideograph-numeric` serializes as `normal` in `getComputedStyle()`. Can you check with @fantasai if that is correct?

Reply:
The reason for that is because `text-autospace: normal` should have the same behavior as `ideograph-alpha ideograph-numeric` but it should serialize as `normal`, so we handle `ideograph-alpha ideograph-numeric` as it would be normal and serialize it as `normal` for avoiding having to represent a distinction between these two ways to set `ideograph-alpha ideograph-numeric`. I will ask if that's ok.

Reply:
I know they have the same behavior, but I do wonder if it's OK in the computed style to alias to `normal`.

Reply:
@mdubet thinks we should serialize the same way as author has declared. I was trying to avoid having states that are equal, but I guess I'll need a separated representation for normal to return the right computed value here. We also need to update the parsing test later.